packages/docs/src/components/navigator.tsx

```diff
@@ -119,15 +132,29 @@ export class Navigator extends React.PureComponent<INavigatorProps, INavigatorSt
         });
     }
 
-    private getMatches() {
-        return filter(this.sections, this.state.query, {
-            key: "filterKey",
-        });
+    public componentDidUpdate() {
+        if (this.shouldCheckSelectedInViewport && this.menuRef != null) {
+            const selectedElement = this.menuRef.querySelector(`.${Classes.INTENT_PRIMARY}`) as HTMLElement;
+            const { offsetTop: selectedTop, offsetHeight: selectedHeight } = selectedElement;
```

Comment:
`const { varName: differentVarName } = obj;` - THIS IS A THING!? I need to read the [TS docs](https://basarat.gitbooks.io/typescript/docs/destructuring.html) more thoroughly.